src/Kanvas/Auth/Services/UserManagement.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Auth\Services;
+
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Kanvas\Apps\Models\Apps;
+use Kanvas\Users\Models\Users;
+
+class UserManagement
+{
+    protected Apps $app;
+
+    /**
+     * Construct function.
+     */
+    public function __construct(
+        protected Users $user
+    ) {
+        $this->app = app(Apps::class);
+    }
+
+    /**
+     * Update current user data with $data
+     *
+     * @param array $data
+     *
+     * @return Users
+     */
+    public function update(array $data): Users
+    {
+        try {
+            $this->user->update(array_filter($data));
+        } catch (ModelNotFoundException $e) {
+            //no email sent
```

Comment:
@arfenis if it fails throw a validation exception or lets discuss what type 
https://github.com/bakaphp/kanvas-ecosystem-api/blob/development/src/Kanvas/Exceptions/ValidationException.php